File: FanfictionDB/modMain.vb
Imports System.IO

Module modMain

    Public cls As clsFanfic
    Public BL As New clsBL
    Public Browser As clsWeb

    Sub ExtractResources()

        Dim path As String = ""

        path = My.Application.Info.DirectoryPath

        If Not File.Exists(path & "\" & "XMLtoINI.xslt") Then
            GetEmbeddedFile("XMLtoINI.xslt")
        End If

        If Not File.Exists(path & "\" & "phantomjs.exe") Then
            GetEmbeddedFile("phantomjs.exe")
        End If

    End Sub

    Sub InitIniFile()

        Dim Desktop As String = Environment.GetFolderPath(Environment.SpecialFolder.Desktop)

        Dim fi As FileInfo

        Dim ifr As IniFileReader

        Dim val As String = ""

        fi = New FileInfo(Application.StartupPath & "\\" & "config.ini")

        If fi.Exists Then
            ifr = New IniFileReader(Application.StartupPath & "\config.ini", True)

            val = ifr.GetIniValue("Output", "Path")

        End If

        If val = "" Then
            val = Desktop
        End If

        BL.OutputPath = val

    End Sub

    Sub Main()

        Dim frmMain As New HtmlGrabber
        Browser = New clsWeb

        InitIniFile()

        frmMain.ShowDialog()

        Browser.Dispose()

    End Sub


End Module

File: FFDownload/HtmlGrabber/modMain.cs
using System;
using System.IO;
using System.Windows.Forms;

namespace HtmlGrabber
{

    static class modMain
    {

        public static clsFanfic cls;
        public static clsBL BL = new clsBL();
        public static clsWeb Browser;

        public static void ExtractResources()
        {

            string path = "";

            path = My.MyProject.Application.Info.DirectoryPath;

            if (!File.Exists(path + @"\" + "XMLtoINI.xslt"))
            {
                modUtility.GetEmbeddedFile("XMLtoINI.xslt");
            }

            if (!File.Exists(path + @"\" + "phantomjs.exe"))
            {
                modUtility.GetEmbeddedFile("phantomjs.exe");
            }

        }

        public static void InitIniFile()
        {

            string Desktop = Environment.GetFolderPath(Environment.SpecialFolder.Desktop);

            FileInfo fi;

            IniFileReader ifr;

            string val = "";

            fi = new FileInfo(Application.StartupPath + @"\\" + "config.ini");

            if (fi.Exists)
            {
                ifr = new IniFileReader(Application.StartupPath + @"\config.ini", true);

                val = ifr.GetIniValue("Output", "Path");

            }

            if (string.IsNullOrEmpty(val))
            {
                val = Desktop;
            }

            BL.OutputPath = val;

        }

        public static void Main(string[] args)
        {

            var frmMain = new HtmlGrabber();
            Browser = new clsWeb();

            InitIniFile();

            frmMain.ShowDialog();

            Browser.Dispose();

        }


    }
}